docs/guides/universal-profile/02-edit-profile.md

```diff
@@ -1,139 +1,222 @@
+
+:::info Learn More
+`links` and `tags` accept an array of objects / strings. So you can add as many as you want!
+
+For `profileImage` and `backgroundImage`, see **Step 1.2** below :arrow_down:
+:::
 
-For your `profileImage` and `backgroundImage`, you will need to add the following informations:
+### 1.2 - Add a profile + background image
 
-- **`width`** / **`height`**: replace the placeholder `Number` by the dimension of the image (in pixels).
-- **`hash`**: use this [tool to generate the **keccak256 hash** of your image](https://emn178.github.io/online-tools/keccak_256_checksum.html).
+:::caution
+Image sizes should be written as numbers, not as strings.
+
+The **max image width** allowed on [universalprofile.cloud](https://universalprofile.cloud) is: `profileImage` <= 800px, `backgroundImage` <= 1800px
+:::
+
+:::info
+The JSON file for LSP3Profile accepts an array of images, so that you images of different sizes / dimensions.
+This way, client interfaces can know which images to pick, based on the container size in their interface.
+:::
+
+For our `profileImage` and `backgroundImage`, we will need to add the following informations:
+
+- **`hash`**: use this **[keccak256 image hash generator](https://emn178.github.io/online-tools/keccak_256_checksum.html)**.
 - **`url`**: upload your images to the LUKSO IPFS Gateway.
 
 Use this [IPFS file uploader tool](https://anarkrypto.github.io/upload-files-to-ipfs-from-browser-panel/public/#) with the settings shown below in green.
 
+- **Node Address:** `api.ipfs.lukso.network`
+- **API Port / Gateway Port:** `443`
+
 ![](./img/ipfs-lukso-settings.jpg)
 
-Drag & Drop your file and click on the **"Upload** button. Once done:
+Drag & Drop your images (you can upload multiple images at once) and click on the **"Upload"** button. Once done:
 
 1. copy the ipfs file identifier (cid) shown in green below.
-2. paste it in the `url` field in your JSON file, after `ipfs://`.
+2. paste it in the `url` field in our JSON file, after `ipfs://`.
 
 ![](./img/ipfs-file-upload.jpg)
 
 Save your JSON file after you have added all your details and images.
 
 > **NB:** don't forget to delete the comments in the JSON file!
 
-You are now ready apply these changes on your Universal Profile. We will see how in the next section :arrow_down:
+We are now ready apply these changes on our Universal Profile. We will see how in the next section :arrow_down:
+
+## Step 2 - Upload our JSON file to IPFS
 
-## Step 2: upload your JSON file to IPFS
+:::note Notice
+The rest of this tutorial should be done in a **new file (`main.js`)**.
+:::
 
-Our tool [lsp-factory.js] provides convenience to upload easily your profile Metadata to IPFS.
+We will now start writing the main code of the tutorial.
+Create a new file `main.js`.
 
-```typescript
-import { LSP3UniversalProfile } from '@lukso/lsp-factory.js';
+Our [lsp-factory.js] tool provides convenience to upload our profile Metadata to IPFS.
```

Comment:
Is there a missing link on `lsp-factory.js` ?

Reply:
@Hugoo no the link is at the bottom of the page ;)

```markdown
[lsp-factory.js]: (../../tools/lsp-factoryjs/deployment/universal-profile.md#uploading-lsp3-metadata-to-ipfs)
[erc725.js]: (../../tools/erc725js/writing-data.md#example)
```